src/web/core/src/main/java/org/geoserver/web/data/resource/BasicResourceConfig.java

```diff
@@ -457,4 +327,133 @@ public void validate(final Form<?> form) {
             }
         }
     }
+
+    public boolean addOtherSRS(ResourceInfo resourceInfo) {
+
+        // first check if its WFS-NG or WMSStore
+        List<String> otherSRS = getOtherSRSFromWFSOrWMS(resourceInfo);
+
+        if (otherSRS != null) {
+            if (!otherSRS.isEmpty()) {
+                resourceInfo
+                        .getMetadata()
+                        .put(FeatureTypeInfo.OTHER_SRS, String.join(",", otherSRS));
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private String getActualNativeSRSCode(ResourceInfo resourceInfo) {
+
+        try {
+            if (resourceInfo instanceof FeatureTypeInfo) {
+                FeatureTypeInfo ft = (FeatureTypeInfo) resourceInfo;
+                return "EPSG:"
+                        + CRS.lookupEpsgCode(
+                                ft.getFeatureSource(null, null).getInfo().getCRS(), false);
```

Comment:
Bizzarre, no guarantee info is properly implemented, it's an API that sees little usage. Better to get the feature type and get the CRS from there instead.

Reply:
@aaime understood will do

Reply:
Yes